src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -908,96 +911,148 @@ private List<Connection> getCongestedConnections() {
+            for (Connection connection: indirectConnections) {
+                SitePinInst oldSinkSpi = connection.getSink();
+                if (!oldSinkSpi.isLUTInputPin() || !oldSinkSpi.isRouted()) {
+                    continue;
+                }
+
+                List<RouteNode> rnodes = connection.getRnodes();
+                RouteNode newSinkRnode = rnodes.get(0);
+                if (newSinkRnode == connection.getSinkRnode()) {
+                    continue;
+                }
+                connection.setSinkRnode(newSinkRnode);
+
+                SitePin newSitePin = newSinkRnode.getNode().getSitePin();
+                String existing = pinSwaps.put(oldSinkSpi, newSitePin.getPinName());
+                assert(existing == null);
+            }
+            LUTTools.swapMultipleLutPins(pinSwaps);
+        }
+
+        if (lutRoutethru) {
+            // When LUT routethru-s are considered, examine both static nets to find
+            // any cases where the *MUX output pin is used as a static source alongside
+            // the *_O output being used as a routethru. In such cases, configure the
+            // OUTMUX* site PIP to source from the 5LUT rather than the default 6LUT
+            // so that no conflict occurs
+            for (Net staticNet : Arrays.asList(design.getGndNet(), design.getVccNet())) {
+                for (SitePinInst spi : staticNet.getPins()) {
+                    if (!spi.isOutPin()) {
+                        continue;
+                    }
+                    SiteInst si = spi.getSiteInst();
+                    if (!Utils.isSLICE(si)) {
+                        continue;
+                    }
+
+                    String pinName = spi.getName();
+                    if (!pinName.endsWith("MUX")) {
                         continue;
                     }
 
-                    List<RouteNode> rnodes = connection.getRnodes();
-                    RouteNode newSinkRnode = rnodes.get(0);
-                    if (newSinkRnode == connection.getSinkRnode()) {
+                    Node muxNode = spi.getConnectedNode();
+                    assert(routingGraph.getPreservedNet(muxNode) == staticNet);
+
+                    Site site = si.getSite();
+                    char lutLetter = pinName.charAt(0);
+                    Node oNode = site.getConnectedNode(lutLetter + "_O");
+                    RouteNode rnode = routingGraph.getNode(oNode);
+                    if (rnode == null || rnode.getOccupancy() == 0) {
+                        // No LUT6 routethru, nothing to be done
                         continue;
                     }
-                    connection.setSinkRnode(newSinkRnode);
 
-                    SitePin newSitePin = newSinkRnode.getNode().getSitePin();
-                    String existing = pinSwaps.put(oldSinkSpi, newSitePin.getPinName());
-                    assert(existing == null);
+                    // Only expect GND net to use SLICE outputs
```

Comment:
Perhaps beyond the scope of this, but there may be some rare cases where VCC is driven from LUTs.

Reply:
Removed this assertion as it wasn't enforcing any functionality.